src/utils.js

```diff
@@ -616,6 +616,69 @@ export const snakeCaseToCamelCase = (snakeCaseStr) => {
         .join('');
 };
 
+/**
+ * Scrolls to the bottom of a page, or until it times out.
+ * @param {Object} page
+ * @param {Number} [maxTimeout=0] How many seconds to scroll for. If 0, will scroll until bottom of page.
+ * @param {Number} [waitForDynamicContent=4] How many seconds to wait for no new content to load before exit.
+ * @returns {Promise}
+ * @memberOf utils
+ */
+export const infiniteScroll = async function (page, maxTimeout = 0, waitForDynamicContent = 4) {
+    let finished;
+    const MAX_TIMEOUT = maxTimeout;
+    const WAIT_FOR_DYNAMIC_CONTENT = waitForDynamicContent;
+    const startTime = Date.now();
+    const maybeResourceTypesInfiniteScroll = ['xhr', 'fetch', 'websocket', 'other'];
+    const resourcesStats = {
+        newRequested: 0,
+        oldRequested: 0,
+        matchNumber: 0,
+    };
+
+
+    page.on('request', (msg) => {
+        if (maybeResourceTypesInfiniteScroll.includes(msg.resourceType())) {
+            resourcesStats.newRequested++;
+        }
+    });
+
+    const checkFinished = setInterval(() => {
+        // console.log(resourcesStats)
+        if (resourcesStats.oldRequested === resourcesStats.newRequested) {
+            resourcesStats.matchNumber++;
+            if (resourcesStats.matchNumber >= WAIT_FOR_DYNAMIC_CONTENT) {
+                clearInterval(checkFinished);
+                finished = true;
+                return;
+            }
+        } else {
+            resourcesStats.matchNumber = 0;
+            resourcesStats.oldRequested = resourcesStats.newRequested;
+        }
+        // check if timeout has been reached
+        if (MAX_TIMEOUT !== 0 && (Date.now() - startTime) / 1000 > MAX_TIMEOUT) {
+            // console.log("Timeout limit reached, exiting infinite scroll.")
+            clearInterval(checkFinished);
+            finished = true;
+        }
+    }, 1000);
```

Comment:
For constants, always use a CONSTANT variable

Reply:
Jan was referring to the `1000`. It should be defined somewhere as `CHECK_FINISHED_INTERVAL_MILLIS` or some such. On the other hand, there's no need to do `const WAIT_FOR_SECS = waitForSecs;`.